src/components/EscDshotDirection/EscDshotDirectionComponent.js

```diff
@@ -0,0 +1,174 @@
+        this._onLoadedCallback = onLoadedCallback;
+        this._motorStopValue = motorStopValue;
+        this._motorSpinValue = motorSpinValue;
+        this._currentSpinningMotor = -1;
+        this._selectedMotor = -1;
+
+        this._contentDiv.load("./components/EscDshotDirection/Body.html", () =>
+        {
+            this._setupdialog();
+        });
+    }
+
+    static get SELECTED_BUTTON_CLASS() { return "selected"; }
+
+    _readDom()
+    {
+        this._domAgreeSafetyCheckBox = $('#motorsEnableTestMode-dialogEscDshotDirection');
+        this._domAgreeButton = $('#escDshotDirectionDialogAgreeButton');
+        this._domMainContentBlock = $('#dialogEscDshotDirectionMainContent');
+        this._domWarningContentBlock = $('#dialogEscDshotDirectionWarning');
+        this._domMixerImg = $('#escDshotDirectionMixerPreviewImg');
+        this._domMotorButtonsBlock = $('#escDshotDirectionSelectMotorButtonsWrapper');
+        this._spinButton = $('#escDshotDirectionCommandSpin');
+        this._buttonRotation1 = $('#escDshotDirectionCommandNormal');
+        this._buttonRotation2 = $('#escDshotDirectionCommandReverse');
+    }
+
+    _setupdialog()
+    {
+        this._readDom();
+        this._createMotorButtons();
+        this._setupSpinButtonEvents();
+        this._setupDirectionButtonEvents();
+        i18n.localizePage();
+
+        this._resetGui();
+
+        this._domAgreeSafetyCheckBox.change(() => {
+            const enabled = this._domAgreeSafetyCheckBox.is(':checked');
+            this._domAgreeButton.toggle(enabled);
+        });
+
+        this._domAgreeButton.click(() => {
+            this._onAgreeButtonClicked();
+        });
+
+
+        const imgSrc = CommonUtils.GetMixerImageSrc(FC.MIXER_CONFIG.mixer, FC.MIXER_CONFIG.reverseMotorDir, FC.CONFIG.apiVersion);
+        this._domMixerImg.attr('src', imgSrc);
+
+        this._onLoadedCallback();
+    }
+
+    _setupDirectionButtonEvents()
+    {
+        this._buttonRotation1.on("click", () => {
+            this._sendCurrentEscSpinDirection(DshotCommand.dshotCommands_e.DSHOT_CMD_SPIN_DIRECTION_1);
+        });
+
+        this._buttonRotation2.on("click", () => {
+            this._sendCurrentEscSpinDirection(DshotCommand.dshotCommands_e.DSHOT_CMD_SPIN_DIRECTION_2);
+        });
+    }
+
+    _sendCurrentEscSpinDirection(direction)
+    {
+        let buffer = [];
+        buffer.push8(this._selectedMotor);
+        buffer.push8(direction);
+        buffer.push8(DshotCommand.dshotCommandType_e.DSHOT_CMD_TYPE_INLINE);
+        MSP.send_message(MSPCodes.MSP2_SEND_DSHOT_COMMAND, buffer);
+
+        buffer = [];
+        buffer.push8(this._selectedMotor);
+        buffer.push8(DshotCommand.dshotCommands_e.DSHOT_CMD_SAVE_SETTINGS);
+        buffer.push8(DshotCommand.dshotCommandType_e.DSHOT_CMD_TYPE_INLINE);
+        MSP.send_message(MSPCodes.MSP2_SEND_DSHOT_COMMAND, buffer);
+    }
+
+    _setupSpinButtonEvents()
+    {
+        this._spinButton.on("mousedown", () => {
```

Comment:
@limonspb We need to add here these atributes for Android compatibility(else it doesnt works) `touchstart` and `touchend` like this:
```
    _setupSpinButtonEvents()
    {
        this._spinButton.on("mousedown touchstart", () => {
            this._spinButton.addClass("selected");
            this._spinMotor(this._selectedMotor);
        });

        this._spinButton.on("mouseup mouseout touchend", () => {
            this._spinButton.removeClass("selected");
            this._spinMotor(-1);
        });
    }
```